descartes_light/solvers/include/descartes_light/solvers/bgl/bgl_ladder_graph_solver.h

```diff
@@ -29,19 +29,13 @@ DESCARTES_IGNORE_WARNINGS_POP
 namespace descartes_light
 {
 /**
- * @brief Graph search leveraging the Boost Graph Library implementation of Dijkstra's algorithm
+ * @brief Partial implementation for solvers leveraging the Boost Graph Library
  */
 template <typename FloatType>
-class BGLLadderGraphSolver : public Solver<FloatType>
+class BGLSolverBase : public Solver<FloatType>
```

Comment:
Should we include `Base` in the name? I think for consistency since the Solver is not named `SolverBase`, it maybe better to name it  `BGLSolver`. I think this would be another good topic for the quality meeting to define a recommend style for naming.

- GBLSolver (Base Class)
- LadderGraphBGLSolver

Reply:
I added the suffix `Base` to these classes because they are intended to be foundations (base) for all the other BGL solvers and they are only partial implementations of the `Solver` interface. I thought adding `Base` might clarify that it can't be used on its own.

I'm ambivalent about the term `LadderGraph` for these solvers. In my opinion, it doesn't really add much more context to the class and might be confusing since the operation of this solver is not related to the other solver that is already called `LadderGraphSolver`. If you feel strongly about keeping the `LadderGraph` term in the BGL class names, I can add it back, but it doesn't seem that valuable to me

Reply:
> I'm ambivalent about the term `LadderGraph` for these solvers. In my opinion, it doesn't really add much more context to the class and might be confusing since the operation of this solver is not related to the other solver that is already called `LadderGraphSolver`. If you feel strongly about keeping the `LadderGraph` term in the BGL class names, I can add it back, but it doesn't seem that valuable to me

Sorry, I was not suggesting adding back `LadderGraph` it was more of naming example. I agree that LadderGraph does not belong in the name.

> I thought adding Base might clarify that it can't be used on its own.
I have also gone back and forth on this type an naming and not sure. In some other packages I decided on removing and just documenting the class, but I do not feel strongly about it. I think this would be good topic for the software quality meeting to define the preferred style.

Reply:
I think having Base is good here because it is not intended to be used in the "Public API" only `Solver`. We should document this style on the confluence page.

Reply:
Sounds good. There is at least documentation on those three classes that they are partial implementations, so if it isn't too obvious from the name the documentation should clarify it